components/browser/state/src/main/java/mozilla/components/browser/state/state/ContentState.kt

```diff
@@ -88,5 +89,6 @@ data class ContentState(
     val loadRequest: LoadRequestState? = null,
     val refreshCanceled: Boolean = false,
     val recordingDevices: List<RecordingDevice> = emptyList(),
-    val desktopMode: Boolean = false
+    val desktopMode: Boolean = false,
+    val appIntent: AppIntentState? = null
```

Comment:
We never had any code that consumed the app intent so we always keep the last one, which seems fine as we only react to changes anyway and we also don't persist it.

Reply:
I wonder if we need to consume now: What if the app gets backgrounded and then comes back to the foreground: Wouldn't we start a new scope and get the last `AppIntentState` again and process it? 🤔

Reply:
Yeah, same. Tested and confirmed that we need it. I've updated the PR and verified that we consume properly now.